syntax = "proto2";

import "nanopb.proto";

package rpc.test;

message getProperty {
    message In {}
    message Out {
        optional float value = 1;
    }
}

message setProperty {
    message In {
        optional float value = 1;
    }
    message Out {}
}

message RpcRequest {
    optional uint32 requestId = 1;

    oneof arg {
        getProperty.In getProperty = 2;
        setProperty.In setProperty = 3;
    }
}

message RpcReply {
    optional uint32 requestId = 1;

    oneof arg {
        getProperty.Out getProperty = 2;
        setProperty.Out setProperty = 3;
    }
}

message Quux {
    optional uint32 value = 1;
}

message ClientToServer {
    oneof arg {
        RpcRequest rpcRequest = 1;
        Quux quux = 2;
    }
}

message ServerToClient {
    oneof arg {
        RpcReply rpcReply = 1;
        Quux quux = 2;
    }
}
